File: frontend/src/types/project.ts
// export interface Project {
//   id: string;
//   workspace_id:string;
//   name: string;
//   description: string;
//   is_public: boolean;
//   // Thêm các trường khác nếu Backend có trả về
// }
export interface Project {
  id: string;
  workspace_id: string;
  name: string;
  description: string | null;
  is_public: boolean;
}

export interface ProjectCreate {
  workspace_id: string;
  name: string;
  description?: string;
  is_public?: boolean;
}

export interface ProjectUpdate {
  name?: string;
  description?: string;
  is_public?: boolean;
}

File: backend/app/schemas/project.py
from uuid import UUID
from typing import Optional
from pydantic import BaseModel

class ProjectCreate(BaseModel):
    workspace_id: UUID
    name: str
    description: Optional[str] = None
    is_public: bool = True

class ProjectUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    is_public: Optional[bool] = None

class ProjectRead(BaseModel):
    id: UUID
    workspace_id: UUID
    name: str
    description: Optional[str]
    is_public: bool

    class Config:
        from_attributes = True